packages/strapi-plugin-content-manager/admin/src/components/SelectWrapper/index.js

```diff
@@ -107,17 +107,13 @@ function SelectWrapper({
       }
 
       return { ...prevState, _q: inputValue };
-    });
-
-    ref.current();
+    }, ref.current);
```

Comment:
Unlike classes the `useState` hook doesn't accept a second argument but here's a way to fix the issue.


1. Remove the abortController from the useEffect hook and remove the arguments of the ref:

```
const abortController = new AbortController();
const { signal } = abortController;

// ...

ref.current = async () => {
  // import cloneDeep from lodash
  const params = cloneDeep(state);
 
  try { /* ... */ } catch(err){}
}
```

2. Use two other useEffect hooks (one for the search and one for the scroll)

```
  // initial one to get the data on the first render
  useEffect(() => {
    ref.current();

    return () => {
      abortController.abort();
    };
  }, [ref]);

  // Fire the request on input change
  useEffect(() => {
    if (state._q !== '') {
      ref.current();
    }

    return () => {
      abortController.abort();
    };
  }, [state._q]);

  // Fire the request when scrolling to the bottom of the list
  useEffect(() => {
    if (state._start !== 0) {
      ref.current();
    }

    return () => {
      abortController.abort();
    };
  }, [state._start]);
```

3. Modify the onChangeInput handler

```
  const onInputChange = (inputValue, { action }) => {
    if (action === 'input-change') {
      setState(prevState => {
        if (prevState._q === inputValue) {
          return prevState;
        }
        return { ...prevState, _q: inputValue };
      });
    }

    return inputValue;
  };
```

4. Modify the onMenuScrollToBottom handler

```
  const onMenuScrollToBottom = () => {
    setState(prevState => ({ ...prevState, _start: prevState._start + 1 }));
  };
```

I believe that with the above modifications fixes the issue.